app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -212,15 +216,26 @@ class SessionControlView(
                 override fun onLayoutCompleted(state: RecyclerView.State?) {
                     super.onLayoutCompleted(state)
 
-                    if (!context.settings().showHomeOnboardingDialog && (
-                        context.settings().showSyncCFR ||
-                            context.settings().shouldShowJumpBackInCFR
-                        )
-                    ) {
-                        HomeCFRPresenter(
-                            context = context,
-                            recyclerView = view,
-                        ).show()
+                    if (!featureRecommended && !context.settings().showHomeOnboardingDialog) {
```

Comment:
Wouldn't `featureRecommended` always be false anyways when this class is created? I think we can avoid this check and simply declare `featureRecommended` inside the if block

Reply:
yes, it will be. but `onLayoutCompleted` is being called multiple times during one lifecycle, in cases like adding items, keyboard, orientation change, and it would launch multiple CFR during one homescreen lifecycle if we don't keep track of CFRs being actually displayed during a single fragment lifecycle

Reply:
I think in general there is a future opportunity to solidify our work around when different onboarding items are shown. That said, I think this solution would work for now given the details @mavduevskiy has listed and our time constraints.